View/ViewHandler.php

```diff
@@ -380,16 +380,26 @@ public function renderTemplate(View $view, $format)
     public function prepareTemplateParameters(View $view)
     {
         $data = $view->getData();
+        $templateData = $view->getTemplateData();
         if ($data instanceof FormInterface) {
-            return array($view->getTemplateVar() => $data->getData(), 'form' => $data->createView());
+            $preparedData = array_merge(
```

Comment:
I would add the following:
```diff
@@ -381,6 +381,9 @@
     {
         $data = $view->getData();
         $templateData = $view->getTemplateData();
+        if (is_callable($templateData)) {
+            $templateData = call_user_func($templateData, $this, $view);
+        }
         if ($data instanceof FormInterface) {
             $preparedData = array_merge(
                 array($view->getTemplateVar() => $data->getData(), 'form' => $data->createView()),
```
At least the `ViewHandlerInterface` (`$this`) and the `View` need to be passed to the callback, so that the format can be read correctly.